dts/bindings/stepper/gpio-stepper-controller.yaml

```diff
@@ -0,0 +1,40 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+# SPDX-License-Identifier: Apache-2.0
+
+description: GPIO Stepper Controller cluster for darlington transistor arrays or dual H-bridge
+  stepper {
+    compatible = "gpio-steppers";
+    status = "okay";
+    motor:motor {
+      gpios = <&gpioa 9 GPIO_ACTIVE_HIGH>,
+              <&gpioc 7 GPIO_ACTIVE_HIGH>,
+              <&gpiob 0 GPIO_ACTIVE_HIGH>,
+              <&gpioa 7 GPIO_ACTIVE_HIGH>;
```

Comment:
how are the coil supposed to be paired here? thought it was AABB but it seems to be ABAB?

Reply:
These are actually  4 different coils namely , a ,b ,c, d

Reply:
done

Reply:
for future: it may be worth using `gpio-names = "A1", "B1", "A2", "B2"`.

Reply:
Yes you are right, on my motor lead A is connected to C and lead B is connected to D, which maps to A1<->A2, B1<->B2 i think on your side, but i will add this to the documentation in yaml

Reply:
so many naming scheme, ok thanks a bunch!